frontend/src/components/student/BadgesPage/BadgesPage.js

```diff
@@ -0,0 +1,163 @@
+import React, { useCallback } from 'react'
+import { Content } from '../../App/AppGeneralStyles'
+import { Col, Row } from 'react-bootstrap'
+import { HorizontalPointsLine, PercentageBar, Tooltip } from './BadgesStyle'
+import { getBadgesInfo, getBadgesList, getLastUnlockedBadge, getRankInfo, getStudentPoints } from './mockData'
+import ContentCard from './ContentCard'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faCircleCheck } from '@fortawesome/free-solid-svg-icons'
+
+function BadgesPage() {
+  const studentRankInfo = getRankInfo()
+  const [studentPoints, missingPoints] = getStudentPoints()
+  const [studentBadgesNumber, allBadgesNumber] = getBadgesInfo()
+  const badgesList = getBadgesList()
+  const lastUnlockedBadge = getLastUnlockedBadge()
+
+  const sortBadges = (a, b) => {
+    if (a.unlocked) {
+      return -1
+    } else return 1
+  }
+
+  const additionalContent = useCallback(
+    (rankIndex, rankMaxPoints) => {
+      const percentageBarWidth = 200
```

Comment:
nit -> jeśli to w sumie stała to może dajmy `PERCENTAGE_BAR_WIDTH`?

Reply:
okej, słusznie :D